### PR TITLE
Use Penny app's token for pushing to template repos

### PR DESCRIPTION
**Pull request by gwynne**
Same idea as vapor/toolbox#416

### CODE REVIEW

**Review by MahdiBM — APPROVED** [review]
LGTM as long as the CI doesn't fail

**Comment by MahdiBM** [comment]
To be honest i expect this to fail ... but if the CI doesn't fail then i'm just wrong

**Comment by MahdiBM** [comment]
Ah i guess i'm wrong, but now that i look more closely into the whole CI file, i'm less surprised to be wrong.

**Review by 0xTim — APPROVED** [review]

File: Package.swift [hunk 1]
```diff
@@ -21,12 +24,6 @@ let package = Package(
                 .product(name: "Fluent{{fluent.db.module}}Driver", package: "fluent-{{fluent.db.url}}-driver"),{{/fluent}}{{#leaf}}
                 .product(name: "Leaf", package: "leaf"),{{/leaf}}
                 .product(name: "Vapor", package: "vapor")
-            ],
-            swiftSettings: [
```
Comment (0xTim) [inline Package.swift:25]: Why are we removing this?

Comment (MahdiBM) [inline Package.swift:25]: Why was this removed? 🤔

**Review by MahdiBM — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (gwynne) [reply Package.swift:25]: @0xTim Per https://www.swift.org/server/guides/building.html#:~:text=cross-module-optimization, "it may sometimes cause performance regressions", and I can't think of any instances I could cite where I saw it have much of _any_ impact. (Also, I believe it's on by default in 5.9, possibly sooner.)

Comment (gwynne) [reply Package.swift:25]: https://github.com/vapor/template/pull/104#discussion_r1267195853 ROFL great minds.

File: Package.swift [hunk 2]
```diff
@@ -4,14 +4,17 @@ import PackageDescription
 let package = Package(
     name: "{{name}}",
     platforms: [
-       .macOS(.v12)
+       .macOS(.v13)
```
[the remark this answers is not in the document]
Comment (0xTim) [reply Package.swift:7]: Yeah I have no problem with this - people can always downgrade if they want

File: Package.swift (hunk 1 above)
Comment (0xTim) [reply Package.swift:25]: I'm conflicted on this - on the one hand performance regressions (for release builds only), on the other hand - no data to prove it either way and it's the current 'recommended' way as per the swift.org guides. Happy to let it slide though
